### PR TITLE
[AOTI] Refactor codegen_inputs in wrapper codegen

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #142136
* #142133
* __->__ #141965
* #141979
* #141387
* #141388

Summary: Fork codegen_inputs for CppWrapperCodegen, because the behavior between python and cpp needs to diverge. On the python side, input backed symbols need to be generated for the autotune block. This is to prepare for one-pass AOTI CUDA codegen.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @chauhang @aakhundov

Differential Revision: [D66718225](https://our.internmc.facebook.com/intern/diff/D66718225)

### CODE REVIEW

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D66718225).

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D66718225).

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D66718225).

File: torch/_inductor/codegen/cpp_wrapper_cpu.py [hunk 1]
```diff
@@ -259,6 +257,40 @@ def write_input_output_info(
...
+
+        @functools.lru_cache(None)
+        def strideof(name):
+            self.codegen_input_stride_var_decl(code, name)
+            return f"{name}_stride"
+
+        if isinstance(value, sympy.Expr):
+            if not isinstance(value, sympy.Symbol) or value in bound_vars:
+                return
+            code.writeline(f"int64_t {value} = {name};")
```
Comment (chenyang78) [inline torch/_inductor/codegen/cpp_wrapper_cpu.py:280]: nit - is this safe to always declare an ```int64_t``` here? If we have any assumption about the type of ```value``` and ```name```, we might want to assert such an assumption?

Comment (desertfire) [reply torch/_inductor/codegen/cpp_wrapper_cpu.py]: We only expect to see symint at the moment, but I have extended it to symfloat following your suggestion.

**Review by chenyang78 — APPROVED** [review]

**Pull request closed without merging**